lib/rl/watcher/compile_format.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Rl.Watcher.CompileFormat do
+  @moduledoc false
+  def init(events) do
+    events
+  end
+
+  def handle_events(events, state) do
+    try do
+      IEx.Helpers.recompile()
+      IO.puts("done compiling")
+    catch
+      _, _ -> :error
+    end
+    |> case do
+      :error ->
+        :error
+
+      _ ->
+        Rl.Watcher.Shell.run("mix format")
+    end
```

Comment:
Don't you want this to be done in reverse? If you compile, then format, it'll trigger another event. If you format first then it should only happen that once.

Reply:
> Whats the purpose of the `Nothing`?
I honestly don't remember, but nothing replaced some shell hacks in our config that were designed to do nothing on certain paths that rl really wanted to watch by running /bin/true, and I figured at least this was faster.

Reply:
> Don't you want this to be done in reverse? If you compile, then format, it'll trigger another event.

Despite looking wrong, I would occasionally end up in compile / format loops with it the other way around.  When I dug deep I figured it out but honestly I don't remember why anymore.  Sorry.